From 64ec94ce6d00f132d27579b954777b4fafec2e57 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Mon, 30 Aug 2010 16:44:53 -0700
Subject: [PATCH 05/26] xen: make pirq interrupts use fasteoi

Use handle_fasteoi_irq for pirq interrupts to match the other Xen
interrupts.

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
(cherry picked from commit 2789ef00cbe2cdb38deb30ee4085b88befadb1b0)

Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
---
 drivers/xen/events.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index 3561d1d..d70e629 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -523,11 +523,11 @@ static void ack_pirq(unsigned int irq)
 {
 	int evtchn = evtchn_from_irq(irq);
 
-	move_native_irq(irq);
-
-	if (VALID_EVTCHN(evtchn))
-		clear_evtchn(evtchn);
 	pirq_eoi(irq);
+	move_masked_irq(irq);
+	
+	if (VALID_EVTCHN(evtchn))
+		unmask_evtchn(evtchn);
 }
 
 static void end_pirq(unsigned int irq)
@@ -592,7 +592,7 @@ int xen_allocate_pirq(unsigned gsi, int shareable, char *name)
 		irq = find_unbound_irq();
 
 	set_irq_chip_and_handler_name(irq, &xen_pirq_chip,
-				      handle_edge_irq, name);
+				      handle_fasteoi_irq, name);
 
 	irq_op.irq = gsi;
 	irq_op.vector = 0;
@@ -1397,7 +1397,7 @@ static struct irq_chip xen_pirq_chip __read_mostly = {
 	.disable	= disable_pirq,
 	.mask		= disable_pirq,
 
-	.ack		= ack_pirq,
+	.eoi		= ack_pirq,
 	.end		= end_pirq,
 
 	.set_affinity	= set_affinity_irq,
-- 
1.7.5.4

